modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/SplitResponseProcessor.java

```diff
@@ -0,0 +1,166 @@
+    ) {
+        super(tag, description, ignoreFailure);
+        this.splitField = Objects.requireNonNull(splitField);
+        this.separator = Objects.requireNonNull(separator);
+        this.preserveTrailing = preserveTrailing;
+        this.targetField = targetField == null ? splitField : targetField;
+    }
+
+    /**
+     * Getter function for splitField
+     * @return sortField
+     */
+    public String getSplitField() {
+        return splitField;
+    }
+
+    /**
+     * Getter function for separator
+     * @return separator
+     */
+    public String getSeparator() {
+        return separator;
+    }
+
+    /**
+     * Getter function for preserveTrailing
+     * @return preserveTrailing;
+     */
+    public boolean isPreserveTrailing() {
+        return preserveTrailing;
+    }
+
+    /**
+     * Getter function for targetField
+     * @return targetField
+     */
+    public String getTargetField() {
+        return targetField;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    @Override
+    public SearchResponse processResponse(SearchRequest request, SearchResponse response) throws Exception {
+        SearchHit[] hits = response.getHits().getHits();
+        for (SearchHit hit : hits) {
+            Map<String, DocumentField> fields = hit.getFields();
+            if (fields.containsKey(splitField)) {
+                DocumentField docField = hit.getFields().get(splitField);
+                if (docField == null) {
+                    throw new IllegalArgumentException("field [" + splitField + "] is null, cannot split.");
+                }
+                Object val = docField.getValue();
+                if (val == null || !String.class.isAssignableFrom(val.getClass())) {
+                    throw new IllegalArgumentException("field [" + splitField + "] is not a string, cannot split");
+                }
+                String[] strings = ((String) val).split(separator, preserveTrailing ? -1 : 0);
+                List<Object> splitList = Stream.of(strings).collect(Collectors.toList());
+                hit.setDocumentField(targetField, new DocumentField(targetField, splitList));
+            }
+            if (hit.hasSource()) {
+                BytesReference sourceRef = hit.getSourceRef();
+                Tuple<? extends MediaType, Map<String, Object>> typeAndSourceMap = XContentHelper.convertToMap(
+                    sourceRef,
+                    false,
+                    (MediaType) null
+                );
+
+                Map<String, Object> sourceAsMap = typeAndSourceMap.v2();
+                if (sourceAsMap.containsKey(splitField)) {
+                    Object val = sourceAsMap.get(splitField);
+                    if (val instanceof String) {
+                        String[] strings = ((String) val).split(separator, preserveTrailing ? -1 : 0);
+                        List<Object> splitList = Stream.of(strings).collect(Collectors.toList());
```

Comment:
```suggestion
                        List<Object> splitList = Arrays.asList(strings);
```

Reply:
Looks like this doesn't work because `Arrays.asList() accepts a varargs param and an array counts:

> SplitResponseProcessor.java:119: warning: non-varargs call of varargs method with inexact argument type for last parameter;
>                 List<Object> splitList = Arrays.asList(strings);
>

Reply:
Actually it's not the varargs, it's the type casting of a list, the same issue discussed at https://github.com/opensearch-project/OpenSearch/pull/14785#discussion_r1684918543

`Arrays.asList(strings)` returns a `List<String>` but we need a `List<Object>` so we have to individually cast.

Reply:
@andrross @owaiskazi19 Coming off of https://github.com/opensearch-project/OpenSearch/pull/14785#discussion_r1685820907 I revisited this one as well.  Unlike lists, arrays are covariant so a simple cast works: this avoids any new array creation:
```java
Object[] strings = ((String) val).split(separator, preserveTrailing ? -1 : 0);
hit.setDocumentField(targetField, new DocumentField(targetField, Arrays.asList(strings)));
```
Which can be shortened to a one-liner that I've committed:
```java
new DocumentField(targetField, Arrays.asList(((String) val).split(separator, preserveTrailing ? -1 : 0)))
```

Reply:
@dbwiddis Looks good. Just to explain my thinking here...I generally try to avoid premature optimizations, but this is hot-path ingestion code so it is worth thinking about minimizing work when doing something at the scale of every document or every field. The Java stream API is very object-creation-happy and can end up creating a lot of work for the garbage collector, whereas when you already have an array the `Arrays.asList` method just ends up creating what is effectively a simple list wrapper object on top of that array without having to copy every element (if I'm reading the code correctly).